recipes/nng/all/test_package/conanfile.py

```diff
@@ -1,17 +1,30 @@
-from conans import ConanFile, CMake, tools
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import cmake_layout, CMake
 import os
 
 
 class TestPackageConan(ConanFile):
-    settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package_multi"
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "CMakeDeps", "CMakeToolchain", "VirtualRunEnv"
+    test_type = "explicit"
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def layout(self):
+        cmake_layout(self)
 
     def build(self):
+        cache = os.path.join(self.build_folder, "CMakeCache.txt")
+        if os.path.exists(cache):
+            self.output.info(f"Removing {cache}")
+            os.remove(cache)
```

Comment:
Because this is a "workflow" problem and it can actually happen with any old version too (like before layouts were added its not specific to that feature)

We do not need a work around. When working with conan 1.x with any generator or options you need to delete the folder.

this sin was fixed in 2.0 and soon that will be the only supported version so it should not be a problem 🤞 

```suggestion
```